pkg/log/zap/logger_test.go

```diff
@@ -166,6 +171,22 @@ func TestGetConfig(t *testing.T) {
 			assert.Equal(t, tc.expected.level, cfg.level)
 			assert.Equal(t, len(tc.expected.opts), len(cfg.opts))
 			assert.Equal(t, tc.expected.sample, cfg.sample)
+
+			dalog := createLogger(cfg, os.Stderr)
+			dalog.V(10).Info("this should not panic")
 		})
 	}
 }
+
+func createLogger(cfg config, dest io.Writer) logr.Logger {
```

Comment:
We could probably put this in `logger.go` and use it from our test and from `LoggerTo` so we don't have to duplicate it.

Reply:
I copied it from logger.go. I'm not 100% sure what all the options are to the logger, I'd probably like to remove a bunch of those. I also don't like putting code that I need in a test in the mainline code base unless it is going to be used generally.  So -1 to putting `createLogger` in `logger.go`

Reply:
Sorry, what I meant was that most of these same lines exist in `LoggerTo`

https://github.com/operator-framework/operator-sdk/blob/720a33546ec1c5d05a1c93e73656fa624b1617ca/pkg/log/zap/logger.go#L33-L47

For `LoggerTo`, we should do something like this:
```go
func LoggerTo(destWriter io.Writer) logr.Logger {
	conf := getConfig()
	return createLogger(config, destWriter)
}

func createLogger(cfg conf, destWriter io.Writer) logr.Logger {
	syncer := zapcore.AddSync(destWriter)
	conf.encoder = &logf.KubeAwareEncoder{Encoder: conf.encoder, Verbose: conf.level.Level() < 0}
	if conf.sample {
		conf.opts = append(conf.opts, zap.WrapCore(func(core zapcore.Core) zapcore.Core {
			return zapcore.NewSampler(core, time.Second, 100, 100)
		}))
	}
	conf.opts = append(conf.opts, zap.AddCallerSkip(1), zap.ErrorOutput(syncer))
	log := zap.New(zapcore.NewCore(conf.encoder, syncer, conf.level))
	log = log.WithOptions(conf.opts...)
	return zapr.NewLogger(log)
}
```

Reply:
I disagree. Because in the test I don't actually set the encoder to a KubeAwareEncoder. Also, while I did copy the other stuff verbatim. I can see us trimming down the call used in the test to something smaller.

Reply:
So I still say -1 to adding a `createLogger` to logger.go. I think they just LOOK the same but I still believe that we will need to be more flexible in how we create the loggers in testing.

Reply:
IMO we should be testing what our public API is using, so I don't think our test should use an alternate method to create the logger. I could set `conf.sample` in `LoggerTo` and break things without tripping the unit test.